modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/planner/StatisticsPlannerTest.java

```diff
@@ -0,0 +1,424 @@
+            10, 1000, t1rc, 8, null, 1, 0));
+
+        colStat1.put("T1C4BYTE", new ColumnStatistics(ValueByte.get((byte)0), ValueByte.get((byte)255),
+            10, 1000, t1rc, 8, null, 1, 0));
+
+        colStat1.put("T1C5BOOLEAN", new ColumnStatistics(ValueBoolean.get(false), ValueBoolean.get(true),
+            0, 2, t1rc, 1, null, 1, 0));
+
+        colStat1.put("T1C6CHARACTER", new ColumnStatistics(ValueString.get("A"), ValueString.get("Z"),
+            10, 10, t1rc, 1, null, 1, 0));
+
+        colStat1.put("T1C7SHORT", new ColumnStatistics(ValueShort.get((short)1), ValueShort.get((short)5000),
+            110, 500, t1rc, 2, null, 1, 0));
+
+        colStat1.put("T1C8LONG", new ColumnStatistics(ValueLong.get(1L), ValueLong.get(100000L),
+            10, 100000, t1rc, 8, null, 1, 0));
+
+        colStat1.put("T1C9FLOAT", new ColumnStatistics(ValueFloat.get((float)0.1), ValueFloat.get((float)0.9),
+            10, 1000, t1rc, 8, null, 1, 0));
+
+        colStat1.put("T1C10DATE", new ColumnStatistics(ValueDate.get(MIN_DATE), ValueDate.get(MAX_DATE),
+            20, 1000, t1rc, 8, null, 1, 0));
+
+        colStat1.put("T1C11TIME", new ColumnStatistics(ValueTime.get(MIN_TIME), ValueTime.get(MAX_TIME),
+            10, 1000, t1rc, 8, null, 1, 0));
+
+        colStat1.put("T1C12TIMESTAMP", new ColumnStatistics(ValueTimestamp.get(MIN_TIMESTAMP), ValueTimestamp.get(MAX_TIMESTAMP),
+            20, 1000, t1rc, 8, null, 1, 0));
+
+        tbl1stat = new IgniteStatisticsImpl(new ObjectStatisticsImpl(1000, colStat1));
+
+        publicSchema = new IgniteSchema("PUBLIC");
+
+        publicSchema.addTable("TBL1", tbl1);
+        publicSchema.addTable("TBL4", tbl4);
+    }
+
+    /**
+     * Check index usage with and without statistics:
+     *
+     * 1) With statistics planner choose second one with better selectivity.
+     * 2) Without statistics planner choose first one.
+     *
+     * @throws Exception In case of error.
+     */
+    @Test
+    public void testIndexChoosing() throws Exception {
+        tbl1.setStatistics(tbl1stat);
+
+        String sql = "select * from TBL1 where t1c7short > 5 and t1c8long > 55555";
+
+        IgniteRel phys = physicalPlan(sql, publicSchema);
+        IgniteIndexScan idxScan = findFirstNode(phys, byClass(IgniteIndexScan.class));
+
+        assertNotNull(idxScan);
+        assertEquals("TBL1_T1C8LONG", idxScan.indexName());
+
+        tbl1.setStatistics(new IgniteStatisticsImpl((GridH2Table)null));
+
+        IgniteRel phys2 = physicalPlan(sql, publicSchema);
+        IgniteIndexScan idxScan2 = findFirstNode(phys2, byClass(IgniteIndexScan.class));
+
+        assertNotNull(idxScan2);
+        assertEquals("TBL1_T1C7SHORT", idxScan2.indexName());
+
+        tbl1.setStatistics(tbl1stat);
+    }
+
+    /**
+     * Check index choosing with is null condition. Due to AbstractIndexScan logic - no index should be choosen.
```

Comment:
```suggestion
     * Check index choosing with is null condition. Due to AbstractIndexScan logic - no index should be chosen.
```

Please fix it here an below